compiler/src/main/java/com/squareup/anvil/compiler/codegen/BindingModuleGenerator.kt

```diff
@@ -473,4 +497,34 @@ private sealed class GeneratedMethod {
   class BindingMethod(override val spec: FunSpec) : GeneratedMethod()
 }
 
+private data class Binding(
+  val contributedClass: ClassDescriptor,
+  val annotation: AnnotationDescriptor,
+  val boundType: ClassDescriptor
+)
+
+private fun List<Binding>.findHighestPriorityBinding(): Binding {
+  if (size == 1) return this[0]
+
+  val bindingsByPriority = groupBy { it.annotation.priority() }
+    .filter { it.value.isNotEmpty() }
+
+  val bindings = bindingsByPriority[HIGHEST]
+    ?: bindingsByPriority[HIGH]
+    ?: bindingsByPriority.getValue(NORMAL)
+
+  if (bindings.size > 1) {
```

Comment:
What is our expected behavior if we have one binding of `HIGH` priority, and two of `NORMAL` one? Right now we'll return the one with `HIGH`, but maybe we want to warn consumer about lower priority clash. Also, it's unclear from the documentation what will be the result.

Reply:
The highest priority wins. That's the only expectation and what's documented. Lower priorities are simply filtered. I don't see the need to print any warnings (Anvil isn't doing this at other places either).